### DISCUSSION TITLE
Hot reloading PNG files

### DISCUSSION THREAD

**Discussion by visualcode-t** [bevyengine/bevy#7071] [Q&A]
I have a very simple program that does not reload textures, when I expect that it would:

```
fn main() {
    App::new()
    .add_plugins(DefaultPlugins.set(AssetPlugin{
        watch_for_changes:true,
        ..Default::default()
    }))
    .add_startup_system(test)
    .run();
}

fn test(mut commands: Commands, assets: Res<AssetServer>) {
    commands.spawn(Camera2dBundle::default());
    commands.spawn(new_image(0.0 ,0.0,3.0,&assets,"img.png"));
}

fn new_image(x: f32, y:f32,z:f32, asset_server: &Res<AssetServer>,img: &str) -> SpriteBundle {
    SpriteBundle {
        texture: asset_server.load(img),
        transform: Transform {
            translation: Vec3::new(x,y,z),
            ..default()
        },
        ..default()
    }
}
```
I added to my Cargo.toml file as well, just in case:

`bevy = {version = "0.9.1", features= ["filesystem_watcher"] }`

What am I doing wrong?

**Comment by frammiie** [comment]
Perhaps try `asset_server.reload_asset` ?
https://docs.rs/bevy/0.9.1/bevy/asset/struct.AssetServer.html#method.reload_asset

**Comment by rparrett** [comment]
Could you give some more details about your environment, and how you're testing this?

The code you provided works fine for me as-is in a new project with
```
bevy = "0.9"
```

(Testing on macOS, just editing the asset in Preview)

**Comment by visualcode-t** [comment]
Sure, I am using Windows 11 22H2x64. I am running the program in debug and attempting to change the graphic in the folder. But, when I do this, it does not update. I do have the following:
```
[profile.dev.package."*"]
opt-level = 3
```

But, I tried without and it failed to change the graphic, too.

**Reply by rparrett** [reply]
What specifically do you mean by "change"?

**Reply by visualcode-t** [reply]
I am attempting to replace the graphic "img.png" in its directory. For example, if I were to rename "img.png" to "img2.png" and name another graphic to "img.png" I would expect that new image to show in my application. Instead, the original graphic remains the only to display.

**Reply by rparrett** [reply]
Thanks. It looks like you're experiencing this issue: <https://github.com/bevyengine/bevy/issues/3359>

**Reply by visualcode-t** [reply]
It is possible that these are related, but I would say that this is different.  I do not receive the error that the file cannot be found. I don't get any indication that it's working at all. However, if I FORCE the reload with `reload_asset` it does change, and it does show an error message if it cannot find the file.

**Reply by rparrett** [reply]
You see the same behavior when modifying the file in-place instead of removing and recreating it?

**Reply by visualcode-t** [reply]
Yes.

**Reply by visualcode-t** [reply]
I tried the Bevy `hot_asset_reloader` and it does work when the gltf file changes. So, either I am doing something wrong with the `png` or it isn't working properly for `png`.

**Comment by visualcode-t** [comment]
Adding a system, does work:

```
fn reload (assets: Res<AssetServer>) {
    assets.reload_asset("Water.png");
}
```

But this runs every cycle.
